src/main/java/teammates/ui/controller/InstructorSearchPageData.java

```diff
@@ -58,4 +139,274 @@ public String getStudentRecordsLink(String courseId, StudentAttributes student){
         link = addUserIdToUrl(link);
         return link;
     }
+    
+    private void setSearchCommentsForStudentsTables() {
+        searchCommentsForStudentsTables = new ArrayList<SearchCommentsForStudentsTable>();       
+        for (String giverEmailPlusCourseId : commentSearchResultBundle.giverCommentTable.keySet()) {
+            searchCommentsForStudentsTables.add(createSearchCommentsForStudentsTable(giverEmailPlusCourseId));
+        }
+    }
+    
+    private void setSearchCommentsForResponsesTables() {
+        searchCommentsForResponsesTables = new ArrayList<SearchCommentsForResponsesTable>();
+        searchCommentsForResponsesTables.add(createSearchCommentsForResponsesTable());
+    }
+    
+    private void setSearchStudentsTables() {
+        searchStudentsTables = new ArrayList<SearchStudentsTable>(); // 1 table for each course      
+        List<String> courseIdList = getCourseIdsFromStudentSearchResultBundle(studentSearchResultBundle.studentList);
+        
+        for (String courseId : courseIdList) {
+            searchStudentsTables.add(createSearchStudentsTable(courseId));
+        }
+    }
+    
+    private List<String> getCourseIdsFromStudentSearchResultBundle(
+                                    List<StudentAttributes> studentList) {
+        List<String> courses = new ArrayList<String>();
+        
+        for (StudentAttributes student : studentSearchResultBundle.studentList) {
+            String course = student.course;
+            if (!courses.contains(course)) {
+                courses.add(course);
+            }
+        }
+        return courses;
+    }
+
+    private SearchCommentsForStudentsTable createSearchCommentsForStudentsTable(String giverEmailPlusCourseId) {
+        String giverDetails = commentSearchResultBundle.giverTable.get(giverEmailPlusCourseId);
+        return new SearchCommentsForStudentsTable(giverDetails, createCommentRows(giverEmailPlusCourseId));
+    }
+    
+    private SearchCommentsForResponsesTable createSearchCommentsForResponsesTable() {
+        return new SearchCommentsForResponsesTable(createFeedbackSessionRows());
+    }
+    
+    private SearchStudentsTable createSearchStudentsTable(String courseId) {
+        return new SearchStudentsTable(courseId, createStudentRows(courseId));
+    }
```

Comment:
I think these one-liner methods can be omitted as they are used only once each and simply calls the appropriate constructor.
@damithc What do you think?